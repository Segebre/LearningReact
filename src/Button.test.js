import React from 'react';
import Renderer from 'react-test-renderer';
import Button from './Button';

it('Button loads correctly',  () => {
    const render = Renderer.create(<Button text="SUBSCRIBE TO BASIC" />);
    const json = render.toJSON();

    expect(json).toMatchSnapshot();
});

it('Button has been clicked', () => {
    const render = Renderer.create(<Button text="SUBSCRIBE TO BASIC" />);
    const button = render.root.findByType('button');

    button.props.onClick();
    expect(button.props.children).toBe('PROCEED TO CHECKOUT')
});