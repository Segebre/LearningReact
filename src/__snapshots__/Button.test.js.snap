// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button loads correctly 1`] = `
<button
  onClick={[Function]}
>
  SUBSCRIBE TO BASIC
</button>
`;
